moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -0,0 +1,199 @@
+ */
+export function useMapDrawTools() {
+  const mapEditorRef = useRef();
+  const [isDrawing, setIsDrawing] = useState(false);
+  const [modeId, setModeId] = useState(null);
+  const [modeHandler, setModeHandler] = useState(null);
+  const [selectedFeatureIndex, setSelectedFeatureIndex] = useState(null);
+  const [selectedEditHandleIndexes, setSelectedEditHandleIndexes] = useState(
+    []
+  );
+
+  /**
+   * Takes the click event and sets the draw mode handler and selected mode ID
+   * @param {object} e - A click event from a draw toolbar button
+   */
+  const switchMode = e => {
+    const switchModeId = e.target.id === modeId ? null : e.target.id;
+    const mode = MODES.find(m => m.id === switchModeId);
+    const modeHandler = mode && mode.handler ? new mode.handler() : null;
+
+    setModeId(switchModeId);
+    setModeHandler(modeHandler);
+  };
+
+  /**
+   * Takes a selected object and sets data about it in state https://github.com/uber/nebula.gl/tree/master/modules/react-map-gl-draw#options
+   * @param {object} selected - Holds data about the selected feature
+   */
+  const onSelect = selected => {
+    setSelectedFeatureIndex(selected && selected.selectedFeatureIndex);
+    setSelectedEditHandleIndexes(
+      selected && selected.selectedEditHandleIndexes
+    );
+  };
+
+  /**
+   * Finds the currently selected feature and removes it from the drawn features array
+   */
+  const onDelete = () => {
+    if (selectedEditHandleIndexes.length) {
+      try {
+        mapEditorRef.current.deleteHandles(
+          selectedFeatureIndex,
+          selectedEditHandleIndexes
+        );
+      } catch (error) {
+        // eslint-disable-next-line no-undef, no-console
+        console.error(error.message);
+      }
+      return;
+    }
+
+    if (selectedFeatureIndex === null || selectedFeatureIndex === undefined) {
+      return;
+    }
+
+    mapEditorRef.current.deleteFeatures(selectedFeatureIndex);
+
+    // Update modeId to momentarily change the background color of the delete icon on click
+    const previousMode = modeId;
+    setModeId("delete");
+    setTimeout(() => setModeId(previousMode), 500);
+  };
+
+  /**
+   * Renders the toolbar and buttons that control the map draw UI
+   * @return {JSX} The toolbar for the map draw UI
+   */
+  const renderDrawToolbar = () => {
+    return (
+      <MapDrawToolbar
+        selectedModeId={modeId}
+        onSwitchMode={switchMode}
+        onDelete={onDelete}
+      />
+    );
+  };
+
+  /**
+   * Renders the map editor and its toolbar
+   * @return {JSX} The whole map draw UI
+   */
+  const renderMapDrawTools = () => (
+    <>
+      <Editor
+        ref={mapEditorRef}
+        featureStyle={getFeatureStyle}
+        onSelect={onSelect}
+        clickRadius={12}
+        mode={modeHandler}
+      />
+      {renderDrawToolbar()}
+    </>
+  );
+
+  return { isDrawing, setIsDrawing, renderMapDrawTools };
+}
```

Comment:
The whole file is amazing, thanks for the docstrings and showing me how you would approach using custom SVGs. It was very easy to follow the code 👨‍💻 ⚡